### DISCUSSION TITLE
Run watch locally

### DISCUSSION THREAD

**Discussion by isolonenko** [TanStack/query#4782] [Q&A] [answered]
Trying to run this project locally. Did the `pnpm install`, switched to the `16.14.2` node ver and trying to run `pnpm run watch` but getting this:

<img width="519" alt="Screenshot 2023-01-09 at 17 47 18" src="https://user-images.githubusercontent.com/103060679/211335583-645209a9-e657-43e3-b410-89a999550f18.png">

macOS: Ventura 13.1

**Comment by TkDodo** [comment, marked as answer]
it works for me, but I've just used it for the first time. There isn't much value in running the rollup build in watch mode ...

maybe the better question is: what do you want to do?

**Reply by isolonenko** [reply]
Just was trying to learn more about the codebase and run this locally to playaround. Therefore was going through contributing.md file to follow the instructions.

Failed at the step mentioned above :)

**Reply by TkDodo** [reply]
😅 which version of pnpm are you using? I'm on `7.12.2`.

What I usually do is run only the tests locally that I am interested in, then make a PR, which creates a preview build environment which I can play around with.

I know @zorzysty has recently updated the readme section on how to run the examples locally:
- https://github.com/TanStack/query/commit/e989725e2732c38f719d1d7d51a3cd0b385adbad

we used to recommend `yalc`, but apparently not anymore. I'm not entirely sure how the examples should pick up the local code even if the local dev server is running, as it only says:

- To run examples, follow their individual directions.
- To run examples using your local build, be sure to have development server (pnpm run watch) running.

@zorzysty can hopefully comment on that

**Reply by isolonenko** [reply]
Thank you. Reinstalled pnpm to 7.12.2 version 🙈

**Reply by TkDodo** [reply]
which version were you on before?
